modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/processing/BlockAcceptanceManager.scala

```diff
@@ -0,0 +1,92 @@
+package org.tessellation.infrastructure.snapshot.processing
+
+import cats.Applicative
+import cats.effect.Async
+import cats.syntax.applicative._
+import cats.syntax.eq._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.show._
+
+import org.tessellation.dag.domain.block.{BlockReference, DAGBlock}
+import org.tessellation.dag.snapshot.BlockAsActiveTip
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.signature.Signed
+
+import monocle.syntax.all._
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+trait BlockAcceptanceManager[F[_]] {
+
+  def processBlocks(
+    initState: BlockAcceptanceState,
+    blocks: List[Signed[DAGBlock]]
+  ): F[BlockAcceptanceResult]
+
+}
```

Comment:
didn't we agree some time ago that such traits we will put into a `domain`? to be able to reference them from services that are in the domain as well 🤔 I remember we had a discussion and that was the output, @TheMMaciek was changing some of the services to split them into domain and infra.

Reply:
Maybe we can have an ADR that would clearly specify what classifies what? It's so ambiguous to me at the moment.

Reply:
yeah, I agree